src/client/src/app/components/users/user-welcome-dialog/user-welcome-dialog.component.ts

```diff
@@ -1,30 +1,62 @@
 import { CommonModule } from '@angular/common';
-import { ChangeDetectionStrategy, Component, inject, signal } from '@angular/core';
+import { ChangeDetectionStrategy, Component, computed, inject, signal } from '@angular/core';
+import { takeUntilDestroyed } from '@angular/core/rxjs-interop';
 import { interpolate, InterpolatePipe } from '@ngneers/signal-translate';
+import { Actions, ofType } from '@ngrx/effects';
+import { Store } from '@ngrx/store';
 import copyToClipboard from 'copy-to-clipboard';
 import { MessageService } from 'primeng/api';
 import { ButtonModule } from 'primeng/button';
 import { DialogModule } from 'primeng/dialog';
+import { ProgressSpinnerModule } from 'primeng/progressspinner';
 
+import { loadUserLoginTokenAction, selectUserLoginToken } from '../../../+state/users';
 import { User } from '../../../models/parsed-models';
 import { TranslateService } from '../../../services/translate.service';
+import { selectSignal } from '../../../utils/ngrx.utils';
 
 @Component({
   selector: 'app-user-welcome-dialog',
   standalone: true,
-  imports: [ButtonModule, CommonModule, DialogModule, InterpolatePipe],
+  imports: [ButtonModule, CommonModule, DialogModule, InterpolatePipe, ProgressSpinnerModule],
   templateUrl: './user-welcome-dialog.component.html',
   changeDetection: ChangeDetectionStrategy.OnPush,
 })
 export class UserWelcomeDialogComponent {
   private readonly _messageService = inject(MessageService);
+  private readonly _store = inject(Store);
   protected readonly translations = inject(TranslateService).translations;
 
   protected readonly visible = signal(false);
   protected readonly user = signal<User | undefined>(undefined);
+  protected readonly isLoading = signal(true);
```

Comment:
Instead of manually setting the loading signal, you can directly select the loading state from the global state:

```suggestion
  private readonly _loadTokenState = selectSignal(selectUsersActionState('loadLoginToken'));
  protected readonly isLoading = computed(() => isActionBusy(this._loadTokenState()));
```